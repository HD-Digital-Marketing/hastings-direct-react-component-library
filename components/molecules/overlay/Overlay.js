import React, {useState} from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import Icon from '../../atoms/icon/Icon'
import classNames from 'classnames'
import './overlay.scss'
import CloseIcon from '../../../src/img/icons/cross.svg'

const Overlay = (props) => {

    const {
        isOpen = false,
        className,
        children
    } = props

    const [isShowing, setIsShowing] = useState(isOpen)

    const toggleOverlay = () => {
        setIsShowing(!isShowing)
    }

    const overlayClass = classNames('m-overlay__wrap', className, {
        'm-overlay__wrap--show': isShowing
    })

    return ReactDOM.createPortal(
        <>
            <div className={overlayClass}>
                <div className="m-overlay__background" onClick={toggleOverlay} />
                <div className="m-overlay">
                    <div className="m-overlay__container">
                        <div className="m-overlay__header">
                            <div>
                                <Icon
                                    svgImage={CloseIcon}
                                    svgWidth={18}
                                    svgHeight={18}
                                    isRound={true}
                                    svgFill="#011831"
                                    svgBorderColor="#60759e"
                                    className="m-overlay__icon"
                                    data-dismiss="modal"
                                    aria-label="Close"
                                    onClick={toggleOverlay} />
                            </div>
                        </div>
                        <div className="m-overlay__content">
                            {children}
                        </div>
                    </div>
                </div>
            </div>
        </>, document.body
    )
}

Overlay.propTypes = {
    isOpen: PropTypes.bool,
    className: PropTypes.string,
    children: PropTypes.node
}

export default Overlay