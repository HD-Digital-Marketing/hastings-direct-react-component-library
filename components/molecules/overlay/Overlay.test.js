import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Overlay from './Overlay'

describe('Check the overlay renders', () => {

    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element, node) => {
            return element
        })
    })

    afterEach(() => {
        ReactDOM.createPortal.mockClear()
    })

    test('it should match the snapshot', () => {

        const tree = renderer
            .create(
                <Overlay isOpen={true}>Overlay content</Overlay>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})