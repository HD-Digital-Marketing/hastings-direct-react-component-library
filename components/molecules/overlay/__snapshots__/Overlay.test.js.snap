// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the overlay renders it should match the snapshot 1`] = `
<div
  className="m-overlay__wrap m-overlay__wrap--show"
>
  <div
    className="m-overlay__background"
    onClick={[Function]}
  />
  <div
    className="m-overlay"
  >
    <div
      className="m-overlay__container"
    >
      <div
        className="m-overlay__header"
      >
        <div>
          <span
            className="a-icon m-overlay__icon a-icon--rounded"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": undefined,
                "borderColor": "#60759e",
                "borderWidth": 1.7999999999999998,
                "height": 36,
                "width": 36,
              }
            }
          />
        </div>
      </div>
      <div
        className="m-overlay__content"
      >
        Overlay content
      </div>
    </div>
  </div>
</div>
`;
